### PR TITLE
Uses exception data to log additional info on the exception message

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs [hunk 1]
```diff
@@ -44,8 +54,11 @@ public async Task<OrchestratorResponse<VacancyInfo>> CloseVacancyAsync(CloseEdit
 
             Utility.CheckAuthorisedAccess(vacancy, m.Ukprn);
 
-            if (!vacancy.CanClose)
-                throw new InvalidStateException(string.Format(ErrorMessages.VacancyNotAvailableForClosing, vacancy.Title));
+            if (!vacancy.CanClose || true)
```
Comment (mrpmorris) [inline src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs]: WTF?

Comment (sachpatel) [reply src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs]: 😱

Comment (dev4valtech) [reply src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs]: This is a draft PR, please comment on the pattern.

Comment (mrpmorris) [reply src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs]: Okay. The pattern of checking `if (X || true)` is bad :)

File: src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs [hunk 2]
```diff
@@ -21,14 +22,23 @@ public CloseVacancyOrchestrator(IProviderVacancyClient client, IRecruitVacancyCl
...
 
             Utility.CheckAuthorisedAccess(vacancy, vrm.Ukprn);
 
             if (!vacancy.CanClose)
-                throw new InvalidStateException(string.Format(ErrorMessages.VacancyNotAvailableForClosing, vacancy.Title));
+            {
+                throw new InvalidStateException(
+                    string.Format(ErrorMessages.VacancyNotAvailableForClosing, vacancy.Title),
+                    user,
+                    new Dictionary<string, string>
```
Comment (chrisjensenuk) [inline src/Provider/Provider.Web/Orchestrators/CloseVacancyOrchestrator.cs]: Maybe we could have an overload where we pass in the `Vacancy` and the `RecruitException` creates the dictionary for us.

**Pull request closed without merging**